### DISCUSSION TITLE
How does that water droplet animation work ?

### DISCUSSION THREAD

**Discussion by PrajjvalMehra** [freshframework/fresh#339] [General]
I am very curious to how that water droplet animation has been achieved

**Comment by lucacasonato** [comment]
It is an SVG that is animated by some client side JavaScript (an island). You can find the source code at https://github.com/denoland/fresh/blob/main/www/islands/LemonDrop.tsx.

@hashrock implemented it, and might have answers if you have more specific questions.

**Comment by hashrock** [comment]
Glad to see someone interested in SVG animation.

![image](https://user-images.githubusercontent.com/3132889/176986051-546d8991-61d6-4ab3-88b3-93936af0f7ab.png)

I first designed this concept in figma. Then I output the SVG and created the lemon part and the WaveTank part as separate SVG elements.

![image](https://user-images.githubusercontent.com/3132889/176985526-9762697a-d943-4a29-89d8-47c5b8ed8ea2.png)
In the lemon part, the function called from requestAnimationFrame edits the d attribute of the path element directly. Simply up and down the control point for this area.

For WaveTank, I referred this article.

https://gamedevelopment.tutsplus.com/tutorials/make-a-splash-with-dynamic-2d-water-effects--gamedev-236

It's very easy to add interactivity, which I didn't include in the LP.

Now we just need to make it look like these two are working together. The only way to do this is to visually adjust the timing, and the code gets more and more jumbled. but it's enjoyable😆
